src/pages/workspaces/workspace/WorkspaceContainer.ts

```diff
@@ -124,6 +126,16 @@ export const WorkspaceContainer = (props: WorkspaceContainerProps) => {
   const isGoogleWorkspaceSyncing =
     workspaceLoaded && isGoogleWorkspace(workspace) && workspace?.workspaceInitialized === false;
 
+  // when the workspace refresh polling gets back an error for a workspace that is deleting
+  // redirect to list view
+  const handleWorkspaceError = (error: unknown) => {
+    if (workspace?.workspace?.state === 'Deleting' && error instanceof Response && error.status === 404) {
```

Comment:
I originally had 501 here as well, because of the cloud context issue.
I changed it to just 404 since we'll eventually get back a 404 if we keep polling, and this way the 501 is just ignored in the meantime.